scripts/colab_helper.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+###############################################################################
+# The Institute for the Design of Advanced Energy Systems Integrated Platform
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES).
+#
+# Copyright (c) 2018-2023 by the software owners: The Regents of the
+# University of California, through Lawrence Berkeley National Laboratory,
+# National Technology & Engineering Solutions of Sandia, LLC, Carnegie Mellon
+# University, West Virginia University Research Corporation, et al.
+# All rights reserved.  Please see the files COPYRIGHT.md and LICENSE.md
+# for full copyright and license information.
+###############################################################################
+"""
+Install IDAES, Ipopt, and other solvers on Google Colab
+
+Created by Alex Dowling (adowling@nd.edu) and Jeff Kantor at the University of Notre Dame
+with input from John Siirola at Sandia National Laboratories.
+
+To use this script, add the following to a code block in a Jupyter notebook:
+
+```
+# Ipopt installer
+import sys
+# If running on Google Colab, install Pyomo and Ipopt via IDAES
+if "google.colab" in sys.modules:
+    !wget "https://raw.githubusercontent.com/idaes-pse/main/scripts/colab_helper.py"
+    import colab_helper
+    colab_helper.install_idaes()
+    colab_helper.install_ipopt()
+```
+
+For testing purposes, you may need to use:
+"https://raw.githubusercontent.com/adowling2/idaes-pse/colab-install-script/scripts/colab_helper.py"
+
+"""
+
+import shutil
+import sys
+import os.path
+import os
+import re
+
+import subprocess
+
+
+def _check_available(executable_name):
+    """Utility to check in an executable is available"""
+    return shutil.which(executable_name) or os.path.isfile(executable_name)
```

Comment:
(More of an FYI than an action) This is find for collab (where we now it is Linux-based).  However, a more portable approach would be to use something akin to Pyomo's `pyomo.common.fileutils.Executable` (which is what Pyomo uses for locating executables)